contracts/staking/ServiceStakingBase.sol

```diff
@@ -0,0 +1,384 @@
+                    uint256 curServiceId = eligibleServiceIds[i];
+                    mapServiceInfo[curServiceId].reward += updatedReward;
+                }
+
+                // If the reward adjustment happened to have small leftovers, add it to the last traversed service
+                if (lastAvailableRewards > totalRewards) {
+                    mapServiceInfo[numServices - 1].reward += lastAvailableRewards - totalRewards;
+                }
+                // Set available rewards to zero
+                lastAvailableRewards = 0;
+            } else {
+                // Traverse all the eligible services and add to their rewards
+                for (uint256 i = 0; i < numServices; ++i) {
+                    // Add reward to the service overall reward
+                    uint256 curServiceId = eligibleServiceIds[i];
+                    mapServiceInfo[curServiceId].reward += eligibleServiceRewards[i];
+                }
+
+                // Adjust available rewards
+                lastAvailableRewards -= totalRewards;
+            }
+
+            // Update the storage value of available rewards
+            availableRewards = lastAvailableRewards;
+        }
+
+        // Updated current service nonces
+        for (uint256 i = 0; i < serviceIds.length; ++i) {
+            // Get the current service Id
+            uint256 curServiceId = serviceIds[i];
+            mapServiceInfo[curServiceId].nonce = serviceNonces[i];
+        }
+
+        // Record the current timestamp such that next calculations start from this point of time
+        tsCheckpoint = block.timestamp;
+
+        emit Checkpoint(lastAvailableRewards, numServices);
+
+        return (serviceIds, numServices, eligibleServiceIds, eligibleServiceRewards);
+    }
+
+    /// @dev Unstakes the service.
+    /// @param serviceId Service Id.
+    function unstake(uint256 serviceId) external {
+        ServiceInfo storage sInfo = mapServiceInfo[serviceId];
+        // Check for the service ownership
+        if (msg.sender != sInfo.owner) {
+            revert OwnerOnly(msg.sender, sInfo.owner);
+        }
+
+        // Call the checkpoint
+        (uint256[] memory serviceIds, , , ) = checkpoint();
+
+        // Get the service index in the set of services
+        // The index must always exist as the service is currently staked, otherwise it has no record in the map
+        uint256 idx;
+        for (; idx < serviceIds.length; ++idx) {
+            if (serviceIds[idx] == serviceId) {
+                break;
+            }
+        }
+
+        // Transfer the service back to the owner
+        IService(serviceRegistry).transferFrom(address(this), msg.sender, serviceId);
+
+        // Send the remaining small balance along with the reward if it is below the chosen threshold
+        uint256 amount = sInfo.reward;
+        uint256 lastAvailableRewards = availableRewards;
+        if (lastAvailableRewards < minBalance) {
```

Comment:
So if into the contract lastAvailableRewards is smaller than minBalance, the first service that is going to call the unstake will receive it

Reply:
yes